plugins/analytics-provider-segment/README.md

```diff
@@ -70,21 +78,3 @@ app:
 
 You might commonly set the above in an `app-config.local.yaml` file, which is
 normally `gitignore`'d but loaded and merged in when Backstage is bootstrapped.
-
-## Development
```

Comment:
Why do we want to move this? We have this as a part of the `README` for other plugins.

Reply:
https://github.com/janus-idp/backstage-plugins/issues/372

Reply:
In that case, this PR forgets to remove those sections from other READMEs. 😉

Reply:
Sorry, I may have misinterpreted some of the development sections as a part of installation setup. The analytics-provider-segment README's development section explicitly mentioned contributing. Meanwhile the other README development sections just discussed how to setup the development environment. 

But now that I reread it, I believe I should also move them to a CONTRIBUTING.md file. I will make the changes now.